### PR TITLE
feat(client): add checkoutSessions clients

### PR DESCRIPTION
**Pull request by boliveira**
## Description

This adds the following changes:

- Adds new clients to manipulate the `checkoutSessions` resource;
- Adds new `SizeOrder` value to `ProductVariantAttributeType` enum;
- Adds missing extensions to linting and format scripts in
`package.json`;
- Fixes metadata type in `CheckoutOrder` to accept only strings and not
other values as the service does not allow other types;

## Checklist

- [X] The commit message follows our guidelines
- [X] Tests for the respective changes have been added
- [X] The code is commented, particularly in hard-to-understand areas
- [X] The labels and/or milestones were added

## Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

File: packages/client/src/checkout/types/getCheckoutSession.types.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import type { CheckoutSession } from './checkoutSession.types.js';
+import type { Config } from '../../types/index.js';
+
+export type GetCheckoutSession = (
+  checkoutOrderId: CheckoutSession['id'],
```
Comment (nelsonleite) [inline packages/client/src/checkout/types/getCheckoutSession.types.ts:5]:
```suggestion
  checkoutSessionId: CheckoutSession['id'],
```

I believe you meant `checkoutSessionId` - is the name of the argument and the same as in `GetCheckoutSessionCharge`

File: packages/client/src/checkout/getCheckoutSession.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { adaptError } from '../helpers/client/formatError.js';
+import client from '../helpers/client/index.js';
+import join from 'proper-url-join';
+import type { GetCheckoutSession } from './types/getCheckoutSession.types.js';
+
+/**
+ * Method responsible for loading the checkout.
+ *
+ * @param checkoutSessionId - If of the checkout session to get.
```
[the remark this answers is not in the document]
Comment (boliveira) [reply packages/client/src/checkout/getCheckoutSession.ts:9]: Thanks, my brain got a segmentation fault clearly when I wrote this XD

File: packages/client/src/checkout/getCheckoutSessionCharge.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { adaptError } from '../helpers/client/formatError.js';
+import client from '../helpers/client/index.js';
+import join from 'proper-url-join';
+import type { GetCheckoutSessionCharge } from './types/getCheckoutSessionCharge.types.js';
+
+/**
+ * Method responsible for getting the orderCharge.
+ *
+ * @param checkoutSessionId - If of the checkout session to get.
```
[the remark this answers is not in the document]
Comment (boliveira) [reply packages/client/src/checkout/getCheckoutSessionCharge.ts:9]: Thanks!

File: packages/client/src/checkout/types/checkoutSessionItem.types.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      CheckoutOrderItem['variants'][number],
+      'formattedPrice' | 'formattedPriceWithoutDiscount' | 'price'
+    > & {
+      price: Omit<
+        CheckoutOrderItem['variants'][number]['price'],
+        | 'formattedPrice'
+        | 'formattedPriceWithoutCurrency'
+        | 'formattedPriceWithoutDiscount'
+        | 'formattedPriceWithoutDiscountAndCurrency'
+        | 'taxType'
```
[the remark this answers is not in the document]
Comment (boliveira) [reply packages/client/src/checkout/types/checkoutSessionItem.types.ts:53]: I was not getting from the endpoint the `taxType` here but I will check again.

File: packages/client/src/checkout/types/getCheckoutSession.types.ts (hunk 1 above)
Comment (boliveira) [reply packages/client/src/checkout/types/getCheckoutSession.types.ts:5]: I meant indeed, thanks for catching that!

File: packages/client/src/checkout/types/checkoutSessionItem.types.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (boliveira) [reply packages/client/src/checkout/types/checkoutSessionItem.types.ts]: Just confirmed, `taxType` is not coming on the `item.variants[].price`. After checking, this behaviour also exists in `CheckoutOrderItem` as well so I updated the types to reflect that. Can you check again? I created another commit with the fixes. Thanks!

**Pull request merged**